include/ensmallen_bits/lmcma/lmcma_impl.hpp

```diff
@@ -0,0 +1,298 @@
+ * the terms of the 3-clause BSD license.  You should have received a copy of
+ * the 3-clause BSD license along with ensmallen.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef ENSMALLEN_LMCMA_LMCMA_IMPL_HPP
+#define ENSMALLEN_LMCMA_LMCMA_IMPL_HPP
+
+#include "lmcma.hpp"
+
+#include <ensmallen_bits/function.hpp>
+
+
+namespace ens {
+    LMCMA::LMCMA(std::size_t N_dim) :
+            T(std::ceil(std::log(N_dim))),
+            c_c(0.5/std::sqrt(N_dim)),
+            c1(1/(10 * std::log(N_dim + 1))),
+            z_bias(0.25),
+            c_sigma(0.3),
+            d_sigma(0.7),
+            m(4 + (size_t)std::floor(3*std::log(N_dim))),
+            lambda(4 + (size_t)std::floor(3*std::log(N_dim))),
+            //lambda(100),
+            w( 1,std::floor( (4 + (size_t)std::floor(3*std::log(N_dim))) / 2 ), arma::fill::zeros),
+            mu(std::floor( (4 + (size_t)std::floor(3 * std::log(N_dim))) / 2 ) )
+    {}
+
+
+    template<typename SeparableFunctionType,
+            typename MatType,
+            typename... CallbackTypes>
+    typename MatType::elem_type LMCMA::optimize(SeparableFunctionType& f,
+                                         MatType& z,
+                                         float sigma,         // TODO: remove from here
+                                         std::size_t n_iter,  // TODO: Remove from here
+                                         CallbackTypes&&... callbacks)
+    {
+
+
+      // Convenience typedefs.
+      typedef typename MatType::elem_type ElemType;
+      typedef typename MatTypeTraits<MatType>::BaseMatType BaseMatType;
+
+      // Make sure that we have the methods that we need.  Long name...
+      traits::CheckArbitrarySeparableFunctionTypeAPI<
+              SeparableFunctionType, BaseMatType>();
+      RequireDenseFloatingPointType<BaseMatType>();
+
+      BaseMatType& iterate = (BaseMatType&) z;
+
+      size_t N_dim = z.n_elem;
+      size_t N = N_dim;
+
+      // Pointer vectors
+      arma::umat J(1, m, arma::fill::zeros);
+      arma::umat L(1, m, arma::fill::zeros);
+
+      // reconstruction step vectors
+      BaseMatType P(N_dim, m, arma::fill::zeros);       // BaseMatType, because is vector in search space
+      BaseMatType V(N_dim, m, arma::fill::zeros);
+      BaseMatType p_c(N_dim, 1, arma::fill::zeros);
+
+      // estimated Expectation values for samples
+      BaseMatType m_old(N_dim, 1, arma::fill::zeros),
+        m_new(N_dim, 1, arma::fill::zeros);
+
+      // weights for estimation of Expectaton value
+      float sum = 0;
+      for(size_t i = 0; i < mu; i++)
+        sum+= log(i+1);
+
+      for(size_t i = 0; i < mu; i++)
+      {
+        float phi = std::log(mu+1) - std::log(i+1);
+        w[i] = (std::log(mu+1) - std::log(i+1) ) / (mu * log(mu + 1) - sum);
+      }
+
+      // const to recompute covariance steps
+      mu_w = 1 / std::pow(arma::norm(w,2), 2);
```

Comment:
Just a style comment that applies throughout... we don't use `_` in variable names, instead preferring camelcase.  So, e.g., this would be `muW`.  See here for more information: https://github.com/mlpack/mlpack/wiki/DesignGuidelines#style-guidelines.  If you don't mind going through there and adjusting your code accordingly, I'd really appreciate it. :)